google/gax/api_callable.py

```diff
@@ -382,6 +382,28 @@ def inner(*args, **kwargs):
     return inner
 
 
+def _merge_options_metadata(options, settings):
```

Comment:
If I recall correctly, there is already some merge logic in `construct_settings` to maintain the right ordering of metadata. Is any of that relevant or reusable?